pkg/providers/subnet/subnet.go

```diff
@@ -136,6 +163,9 @@ func (p *DefaultProvider) ZonalSubnetsForLaunch(ctx context.Context, nodeClass *
 		if ips, ok := p.inflightIPs[*subnets[j].SubnetId]; ok {
 			jIPs = ips
 		}
+		if iIPs == jIPs {
+			return lo.FromPtr(subnets[i].SubnetId) < lo.FromPtr(subnets[j].SubnetId)
+		}
 		return iIPs < jIPs
 	})
 	for _, subnet := range subnets {
```

Comment:
I wonder if rather than sorting first and then iterating through everything and assigning the top subnet to the AZ if we just iterate once and then check if the existing subnet has more IPs. If you do this you should also be able to just use a `map[string]int64` to store the IP address count since you can just grab it out of the map when you are iterating through the subnets rather than building a separate data structure

Reply:
But we would only want to track it for the top subnet in a zone, not for all subnets in every zone. I'm not seeing how this would be tracked?

Reply:
But we expect the zonalSubnets to contain subnets that have the most available IPs, If we don't sort first including the inflight IPs, how would know which subnet to pick for each zone? From my understand the above code, just adds all the subnets in the zonalSubnets map